### DISCUSSION TITLE
can i store rich text file(might include img vid) in public folder ?

### DISCUSSION THREAD

**Discussion by NewUniverseEntrop** [vercel/next.js#15089] [Help] [answered]
Thank you ! 

is it good practice to  store large/big  file in public folder ? for example when user upload a img file or rich text file i mentioned or shall  i use cloud storage instead ?  

i was wondering if public folder is  viewable on client side(is it on vercel's server , i might  store private data on public folder 

(btw, how much does vercel charge for storage ? i couldnt find  ) 
tks !

**Comment by timneutkens** [comment, marked as answer]
The public folder won't be writeable, our general recommendation is using an external data store for large files, for example s3 or google cloud storage.

**Reply by NewUniverseEntrop** [reply]
ah wow, i have been writing image file to the public folder on my local host and it worked fine ...

**Reply by timneutkens** [reply]
Yeah that won't work when deployed to an auto-scaling serverless solution given that you don't have direct filesystem access (the filesystem is readonly).

**Reply by NewUniverseEntrop** [reply]
Thanks, i just tested it and you are definitly right ! it would be awesome  would if it allows write access lol , maybe link public folder with external cloud storage
